columnflow/tasks/framework/mixins.py

```diff
@@ -1852,34 +1767,44 @@ def resolve_param_values(cls, params: dict[str, Any]) -> dict[str, Any]:
 
         return params
 
-    @classmethod
-    def get_inference_model_inst(cls, inference_model: str, config_inst: od.Config) -> InferenceModel:
-        return InferenceModel.get_cls(inference_model)(config_inst)
-
     @classmethod
     def req_params(cls, inst: law.Task, **kwargs) -> dict:
         # prefer --inference-model set on task-level via cli
         kwargs["_prefer_cli"] = law.util.make_set(kwargs.get("_prefer_cli", [])) | {"inference_model"}
 
         return super().req_params(inst, **kwargs)
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-        # get the inference model instance
-        self.inference_model_inst = self.get_inference_model_inst(self.inference_model, self.config_inst)
-
     @property
     def inference_model_repr(self):
         return str(self.inference_model)
 
     def store_parts(self) -> law.util.InsertableDict:
+        """
+        :return: Dictionary with parts that will be translated into an output directory path.
+        """
         parts = super().store_parts()
         if self.inference_model != law.NO_STR:
-            parts.insert_before("version", "inf_model", f"inf__{self.inference_model_repr}")
+            parts.insert_after(self.config_store_anchor, "inf_model", f"inf__{self.inference_model_repr}")
         return parts
 
 
+class InferenceModelMixin(InferenceModelClassMixin):
```

Comment:
does this mixin not need a `inference_model_inst` parameter?

Reply:
Indeed. I generalized the `TaskArrayFunction(s)Parameter` to `DerivableInst(s)Parameter` to be able to capture Derivable subclasses as well and added it :+1: